streams/src/test/java/org/apache/kafka/streams/integration/KStreamTransformIntegrationTest.java

```diff
@@ -55,61 +60,93 @@
     private final String topic = "stream";
     private final String stateStoreName = "myTransformState";
     private final List<KeyValue<Integer, Integer>> results = new ArrayList<>();
-    private final ForeachAction<Integer, Integer> action = (key, value) -> results.add(KeyValue.pair(key, value));
+    private final ForeachAction<Integer, Integer> accumulateExpected = (key, value) -> results.add(KeyValue.pair(key, value));
```

Comment:
Not your fault, but I couldn't figure out how this test actually works. It doesn't look like we ever clear `results` between runs. What's going on here?

Reply:
I would assume it's a JUnit test that uses the [standard lifecycle](https://stackoverflow.com/a/19381563/1031507) where the class is recreated for every test.  I haven't looked into the test running infrastructure at all though so maybe because this is `@Category({IntegrationTest.class})` you would expect some other behavior?

Reply:
Huh, I never knew that. After reading into it a little more, I can see why it works, but it's also not guaranteed to work for all Runner implementations; the docs just say this is the behavior of the _default_ implementation.

Anyway, like I said, it predates your PR, so I wouldn't request any changes. I just thought you might have some insight since you had modified the test. And you did! ;)

Thanks.